controllers/osbuild_controller.go

```diff
@@ -47,16 +82,254 @@ type OSBuildReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
 func (r *OSBuildReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx).WithValues("osbuild", req.Name)
+
+	osBuild, err := r.OSBuildRepository.Read(ctx, req.Name, req.Namespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{Requeue: true, RequeueAfter: requeueForShortDuration}, err
+	}
+
+	if osBuild.DeletionTimestamp != nil {
+		// The OSBuild CRs that were created by that OSBuildConfig would be deleted
+		// thanks to setting controller reference for each OSBuild CR
+		return ctrl.Result{}, nil
+	}
+
+	lastBuildStatus := osbuildv1alpha1.OSBuildConditionType("")
+	if osBuild.Status.Conditions != nil {
+		conditionLen := len(osBuild.Status.Conditions)
+		lastBuildStatus = osBuild.Status.Conditions[conditionLen-1].Type
+	}
+
+	if osBuild.Status.ContainerComposeId == emptyComposeID {
+		// if the edge container wasn't created yet - schedule a new build
+		logger.Info("create an edge-container")
+		err = r.postComposeEdgeContainer(ctx, logger, osBuild)
+		if err != nil {
+			logger.Info("failed to create an edge-container")
+			return ctrl.Result{Requeue: true, RequeueAfter: requeueForLongDuration}, err
+		}
+
+		logger.Info("new job created for edge-container, requeue to sample its status")
+		return ctrl.Result{Requeue: true, RequeueAfter: requeueForLongDuration}, nil
+	} else if lastBuildStatus == startedContainerBuild {
```

Comment:
I don't think so: 
If in the current iteration  ContainerComposeId is not empty and the `lastBuildStatus` is finished we want to go to the next if-else (for building an edge installer job )

Reply:
Still, whenever you have an `if` branch that always returns, there is no need for `else`. I didn't say they is no need for another `if`, just that there is no need for `else if`. The `else` protects your code from entering that branch after the `if` section is done. But, since you always return, the protection is not needed

Reply:
got it now